### ISSUE TITLE
new column names fro as.data.frame.brmsfit()

### ISSUE THREAD

**Issue by strengejacke** [paul-buerkner/brms#827]
Hi Paul,
the latest brms-update introduced some breaking changes, which affect functionality of some of our packages. It looks like the returned column names are no longer "transformed" using `make.names()`:

``` r
m <- insight::download_model("brms_zi_3")
#> Registered S3 method overwritten by 'xts':
#>   method     from
#>   as.zoo.xts zoo
summary(m)
#> Warning: There were 10 divergent transitions after warmup. Increasing adapt_delta above 0.8 may help.
#> See http://mc-stan.org/misc/warnings.html#divergent-transitions-after-warmup
#>  Family: zero_inflated_poisson 
#>   Links: mu = log; zi = logit 
#> Formula: count ~ child + camper + (1 | persons) 
#>          zi ~ child + camper + (1 | persons)
#>    Data: zinb (Number of observations: 250) 
#> Samples: 1 chains, each with iter = 500; warmup = 250; thin = 1;
#>          total post-warmup samples = 250
#> 
#> Group-Level Effects: 
#> ~persons (Number of levels: 4) 
#>                  Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
#> sd(Intercept)        1.68      0.98     0.64     4.68 1.01       65       42
#> sd(zi_Intercept)     1.75      0.94     0.48     3.98 1.01       68      138
#> 
#> Population-Level Effects: 
#>              Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
#> Intercept        1.19      0.84    -0.66     2.84 1.00       88      102
#> zi_Intercept    -0.73      0.74    -2.03     0.86 1.01       90       67
#> child           -1.16      0.11    -1.40    -0.97 1.04      177      146
#> camper           0.73      0.09     0.59     0.91 1.00      233      195
#> zi_child         1.88      0.31     1.27     2.48 1.02       65      118
#> zi_camper       -0.84      0.34    -1.48    -0.22 1.00      184       83
#> 
#> Samples were drawn using sampling(NUTS). For each parameter, Eff.Sample 
#> is a crude measure of effective sample size, and Rhat is the potential 
#> scale reduction factor on split chains (at convergence, Rhat = 1).

# new since brms 2.11
colnames(as.data.frame(m))
#>  [1] "b_Intercept"                "b_zi_Intercept"            
#>  [3] "b_child"                    "b_camper"                  
#>  [5] "b_zi_child"                 "b_zi_camper"               
#>  [7] "sd_persons__Intercept"      "sd_persons__zi_Intercept"  
#>  [9] "r_persons[1,Intercept]"     "r_persons[2,Intercept]"    
#> [11] "r_persons[3,Intercept]"     "r_persons[4,Intercept]"    
#> [13] "r_persons__zi[1,Intercept]" "r_persons__zi[2,Intercept]"
#> [15] "r_persons__zi[3,Intercept]" "r_persons__zi[4,Intercept]"
#> [17] "lp__"

# colnames until brms 2.10
make.names(colnames(as.data.frame(m)))
#>  [1] "b_Intercept"                "b_zi_Intercept"            
#>  [3] "b_child"                    "b_camper"                  
#>  [5] "b_zi_child"                 "b_zi_camper"               
#>  [7] "sd_persons__Intercept"      "sd_persons__zi_Intercept"  
#>  [9] "r_persons.1.Intercept."     "r_persons.2.Intercept."    
#> [11] "r_persons.3.Intercept."     "r_persons.4.Intercept."    
#> [13] "r_persons__zi.1.Intercept." "r_persons__zi.2.Intercept."
#> [15] "r_persons__zi.3.Intercept." "r_persons__zi.4.Intercept."
#> [17] "lp__"
```

<sup>Created on 2020-01-13 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Is this change intentional, so we can revise our packages, or would you revert this change?

**Comment by strengejacke** [comment]
Ok, I _think_ the changes are due to this commit: https://github.com/paul-buerkner/brms/commit/7111ba5656db1c7618c11e848f20266b18530d61

And the title suggests some intention behind this change. I think the fix is rather easy, which will work both for older and the new brms, just by using the `optional`-argument.

**Comment by paul-buerkner** [comment]
This change was intentional as it was more of a bug to use `make.names` in `as.data.frame` rather than expected behavior. The change didn't cause any changes to worse in revesere dependencies (at least what I saw) and so I didn't think it affected any other packages.

**Comment by strengejacke** [comment]
>  The change didn't cause any changes to worse in revesere dependencies (at least what I saw) and so I didn't think it affected any other packages.

We excluded some more time-consuming checks from running on CRAN, that's why we first detect them when it's "too late" ;-) And even if we did not exclude those tests: all packages we test our functions with, are "suggested" packages, and those are not tested in rev-dep checkds, either.

Just wanted to make sure the change is intentional. Fix is rather easy, so I'll close this now.
